Composer/packages/client/src/pages/language-generation/code-editor.tsx

```diff
@@ -2,45 +2,101 @@
 // Licensed under the MIT License.
 
 /* eslint-disable react/display-name */
-import React, { useState, useEffect } from 'react';
-import { LgEditor } from '@bfc/code-editor';
+import React, { useState, useEffect, useMemo, useContext } from 'react';
+import { LgEditor, LGOption } from '@bfc/code-editor';
 import get from 'lodash/get';
 import debounce from 'lodash/debounce';
 import isEmpty from 'lodash/isEmpty';
-import { CodeRange, LgFile } from '@bfc/shared';
+import { Diagnostic } from 'botbuilder-lg';
+import { LgFile } from '@bfc/shared';
 
+import { StoreContext } from '../../store';
 import * as lgUtil from '../../utils/lgUtil';
 
 interface CodeEditorProps {
   file: LgFile;
-  onChange: (value: string) => void;
-  codeRange?: Partial<CodeRange> | null;
+  template: lgUtil.Template | null;
 }
 
+const lspServerPort = process.env.LANGUAGE_SERVER_PORT ? Number(process.env.LANGUAGE_SERVER_PORT) : 5000;
```

Comment:
i forgot we were using the existing express server listening on `3000` to include the websocket endpoint. Shouldn't this _have_ to be whatever the server port is? 

`process.env.PORT`

I'm seeing errors in the dev tools when trying to connect to 5000
<img width="1680" alt="Screen Shot 2019-11-24 at 8 49 04 PM" src="https://user-images.githubusercontent.com/1156704/69513200-dfa60100-0efb-11ea-8a23-0dbd3bd101b9.png">

Reply:
updated to
```
const lspServerPort = process.env.NODE_ENV === 'production' ? process.env.PORT || 3000 : 5000;
```